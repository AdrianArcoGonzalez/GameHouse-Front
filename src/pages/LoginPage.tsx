import { ToastContainer } from "react-toastify";
import heroTexts from "../components/HeroSection/heroData/heroData";
import HeroSection from "../components/HeroSection/HeroSection";
import Login from "../components/Login/Login";
import "react-toastify/dist/ReactToastify.css";
import { useEffect } from "react";

const LoginPage = (): JSX.Element => {
  useEffect(() => {
    window.scroll({ top: 0 });
  }, []);
  return (
    <>
      <ToastContainer />
      <HeroSection
        text={heroTexts.signIn}
        srcImage={"images/singindarksouls.webp"}
      />
      <Login />;
    </>
  );
};

export default LoginPage;
