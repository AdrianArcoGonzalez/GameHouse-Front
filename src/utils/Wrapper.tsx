import * as reactRedux from "react-redux";
import { store } from "../store/store";

export interface WrapperProps {
  children: JSX.Element | JSX.Element[];
}

export const Wrapper = ({ children }: WrapperProps): JSX.Element => {
  return <reactRedux.Provider store={store}>{children}</reactRedux.Provider>;
};
