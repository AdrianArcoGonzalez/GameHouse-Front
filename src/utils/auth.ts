import jwt from "jwt-decode";

const decodeToken = (token: string) => {
  const payloadToken: {
    id: string;
    username: string;
    iat: number;
  } = jwt(token);

  const user = {
    token: token,
    id: payloadToken.id,
    username: payloadToken.username,
    isLogged: true,
  };
  return user;
};

export default decodeToken;
