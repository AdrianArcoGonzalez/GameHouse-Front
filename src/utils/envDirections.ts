export const backUrl = process.env.REACT_APP_URL_BACK

export const urlsBack = {
  default: `${backUrl}games/games/`,
  getByOwner: `${backUrl}games/games/my-collection/`,
  getByCategory: `${backUrl}games/games/category/`,

}
