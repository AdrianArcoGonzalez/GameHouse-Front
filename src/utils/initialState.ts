import decodeToken from "./auth";

const emptyUser = {
  id: "",
  username: "",
  token: "",
  isLogged: false,
};
const initialState = () => {
  const token = localStorage.getItem("token") as string;

  if (token) {
    return decodeToken(token);
  }

  return emptyUser;
};

export default initialState;
