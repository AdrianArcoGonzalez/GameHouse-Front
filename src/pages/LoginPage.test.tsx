import { Wrapper } from "../utils/Wrapper";
import LoginPage from "./LoginPage";
import TestRenderer from "react-test-renderer";

describe("Given a Login Page component", () => {
  describe("When it's instantiated", () => {
    test("Then it should match the the Login page screenshot", () => {
      const expectedLoginPage = TestRenderer.create(
        <Wrapper>
          <LoginPage />
        </Wrapper>
      );

      expect(expectedLoginPage).toMatchSnapshot();
    });
  });
});
