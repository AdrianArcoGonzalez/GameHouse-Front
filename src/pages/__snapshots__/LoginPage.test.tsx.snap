// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Register Page component When it's instantiated Then it should render the Register page 1`] = `
Array [
  <section
    className="sc-bczRLJ ggGwXn"
    data-testid="section"
  >
    <img
      alt="backgroundImage"
      className="hero__image"
      src="images/singindarksouls.webp"
    />
    <div
      className="hero-container"
    >
      <p
        className="hero-container__sentence"
      >
        A strong man doesn’t need to read the future. He makes his own.” (Solid Snake)
      </p>
    </div>
  </section>,
  <form
    autoComplete="off"
    className="sc-gsnTZi dWgaqc"
    data-testid="form"
    noValidate={true}
    onSubmit={[Function]}
  >
    <h2
      className="login__title"
    >
      Sign In
    </h2>
    <label
      className="form__input-container"
      htmlFor="username"
    >
      Username
    </label>
    <input
      className="form__input-element"
      id="username"
      onChange={[Function]}
      placeholder="Username"
      type="text"
      value=""
    />
    <label
      className="form__input-container"
      htmlFor="password"
    >
      Password
    </label>
    <input
      className="form__input-element"
      id="password"
      onChange={[Function]}
      placeholder="*******"
      type="password"
      value=""
    />
    <button
      className="form-button"
      disabled={true}
      type="submit"
    >
      Sign In
    </button>
    <span
      className="form__info"
    >
      Haven't an account yet? Go to
       
      <a
        className="info__link"
        href="/register"
        onClick={[Function]}
      >
        Sign Up
      </a>
    </span>
  </form>,
  ";",
]
`;
